### PR TITLE
replace vendored libcurl and minhook with submodules

### PR DESCRIPTION
**Pull request by Jan200101**
minhook uses the latest commit since CMake support is relatively new.

This was compiled against msvc-wine and tested using the Parkour gamemode and a round of attrition.

Only libcurl and minhook have been turned into submodules, since those are the only ones we need to actually compile.

closes #414

### CODE REVIEW

**Comment by pg9182** [comment]
`ghcr.io/pg9182/northstar-dedicated-ci:1-tf2.0.11.0-ns1.16.0-ci.c14bdc62.100.5436795521-nslauncher.242fa84c.5436737074`

File: cmake/Findspdlog.cmake [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+if(NOT spdlog_FOUND)
+	add_subdirectory(${PROJECT_SOURCE_DIR}/thirdparty/spdlog spdlog)
+    set(spdlog_FOUND 1 PARENT_SCOPE)
```
Comment (pg9182) [inline cmake/Findspdlog.cmake:4]: indentation

**Review by pg9182 — APPROVED** [review]
- Tested in Docker on a Fedora 38 host, an Ubuntu 22.04 host, and an Alpine 3.18 host. Game launches, registers, and successfully accepts players.
- Submodule configuration is okay.
- CMake also looks good.
- libcurl build options are fine.

**Comment by GeckoEidechse** [comment]
Reverted the PR due to missing doc updates and CI changes. Nothing wrong with the PR inherently, it just wasn't ready yet and shouldn't have been merged at this stage. Sorry about all of that, and thanks for making new PR <3

**Pull request merged**